tensorflow_quantum/python/differentiators/linear_combination.py

```diff
@@ -92,16 +92,77 @@ def __init__(self, weights, perturbations):
                              "length.")
         if not len(list(set(perturbations))) == len(perturbations):
             raise ValueError("All values in perturbations must be unique.")
-        self.weights = tf.constant(weights)
+        if len(perturbations) < 2:
+            # This is so that tensor squeezing does not cause a problem later.
+            raise ValueError("Must specify at least two perturbations.")
+        self.weights = tf.constant(weights, dtype=tf.float32)
```

Comment:
Is this a new error mechanism ? Did we always require the user to have 2 perturbations ?

Reply:
Having only one perturbation is the same as evaluating the circuit once at an offset, there need to be at least two evaluation points for differentiation.  There wasn't a limit before; I've now expanded the error message.

In particular, inside ForwardDifference we have `grid_points_to_eval = np.arange(0, error_order + 1)`, and `error_order` is required to be greater than 0, so it also already required 2 or more eval points.  Does this make sense?

Reply:
Ahhhh that does make sense!

In that case I think all we need is a few tests for these new error mechanisms and we're good to merge.

Reply:
There is a test now checking that at least two perturbations are specified